presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -120,7 +121,9 @@ private boolean exploreNode(int group, Context context)
             }
 
             done = true;
-            for (Rule rule : ruleStore.getCandidates(node)) {
+            Iterator<Rule> possiblyMatchingRules = ruleStore.getCandidates(node).iterator();
```

Comment:
Please squash this commit with the previous one, there's no point in introducing it one way and then doing this simple refactoring.